File: Back/src/main/java/com/link/back/dto/response/SkillDto.java
package com.link.back.dto.response;

import com.link.back.entity.Skill;
import com.link.back.entity.UserSkill;

import lombok.Builder;
import lombok.Getter;

@Getter
public class SkillDto {

	private final String skillName;
	private final int SkillLevel;
	private final String skillImageUrl;
	private final String skillType;

	public SkillDto(UserSkill userSkill) {
		Skill skill = userSkill.getSkill();
		this.skillName = skill.getSkillName();
		this.SkillLevel = userSkill.getSkillLevel();
		this.skillImageUrl = skill.getSkillImageUrl();
		this.skillType = skill.getSkillType().getName();
	}

}

File: Front/src/dto/tmpDTOs/skillDTO.ts
export interface SkillDTO {
    skillId:number,
    skillName:string,
    skillImgUrl:string,
    skillYear:number
}

export interface SkillDto {
    skillName: string;
    skillImageUrl: string;
    skillLevel: number;
}

export interface SkillDtoT {
    skillName: string;
    skillImageUrl: string;
    skillLevel: number;
    skillType: string;
}

export interface SkillResponseDto {
    skillName: string;
    skillImageUrl: string;
    skillLevel: string;
    skillType: string;
}

//사용자 정보 추가 등록, 갱신에 사용
export interface SkillRequestDto {
    skillId: number;
    skillName: string;
    skillImageUrl: string;
    skillType: string;
}